proto/server.proto

```diff
@@ -140,6 +140,37 @@ message SignUpRequest {
   Meta meta = 5;
 }
 
+message CreateFolderRequest {
+  string path = 1;
+  string name = 2;
+  Meta meta = 3;
+}
+message ListFilesRequest {
+  string path = 1;
+  Meta meta = 2;
+}
+message DeleteFileRequest {
+  string path = 1; // Full path, including the file name and extension
+  Meta meta = 2;
+}
+message DownloadFileRequest {
+  string path = 1; // Full path, including the file name and extension
+  Meta meta = 2;
+}
+message UploadFileRequest {
+  string path = 1;
+  string name = 2;
+  bytes payload = 3;
+  int64 size = 4; // Size of the whole file in bytes
+  string hash = 5; // Hash for the whole file
```

Comment:
How are you planing to use the hash? If the file is too large it won't fit in the RAM. How will you hash then?

Reply:
I was thinking that we store the file, and then check the hash.

Reply:
Let's add that in a future release? You can open an issue for it. Let's not do hash of now